packages/roosterjs-content-model/lib/editor/plugins/PastePlugin/utils/linkParser.ts

```diff
@@ -0,0 +1,39 @@
+import { ContentModelHyperLinkFormat } from '../../../../publicTypes/format/ContentModelHyperLinkFormat';
+import { FormatParser } from '../../../../publicTypes/context/DomToModelSettings';
+import { safeInstanceOf } from 'roosterjs-editor-dom';
+
+const HTTP = 'http:';
+const HTTPS = 'https:';
+const NOTES = 'notes:';
+
+const sanitizeLinks: FormatParser<ContentModelHyperLinkFormat> = (
+    format,
+    element,
+    context,
+    defaultStyle
+) => {
+    if (!safeInstanceOf(element, 'HTMLAnchorElement')) {
+        return;
+    }
+
+    let url: URL | undefined;
+    try {
+        url = new URL(element.href);
+    } catch {
+        url = undefined;
+    }
+
+    if (
+        !url ||
+        !(
+            url.protocol === HTTP ||
+            url.protocol === HTTPS ||
+            url.protocol === NOTES
```

Comment:
There are requests to allow more protocols, such as onenote.

We can open bug to discuss what should we do here. For now we can keep going with this.

Reply:
Added an issue for this requirement